CANTroller2/src/cantroller2.cpp

```diff
@@ -199,9 +201,13 @@ void loop() {
 
     if (timestamp_loop) loop_savetime (looptimes_us, loopindex, loop_names, loop_dirty, "pre");
 
-    if (take_temperatures) temp_soren();
-    if (simulator.can_simulate(SimOption::coolant) && simulator.get_pot_overload() == SimOption::coolant) temps_f[ENGINE] = pot.mapToRange(temp_sensor_min_f, temp_sensor_max_f);
-    
+    if (take_temperatures) temperature_sensor_manager.update_temperatures();
+    if (simulator.can_simulate(SimOption::coolant) && simulator.get_pot_overload() == SimOption::coolant) {
```

Comment:
so we should make this change in a different PR, but i think you could pretty easily convert the temp sensor manager to be a device (it has a pin, it responds to ControllerStates, can be overloaded by the pot, etc). the advantage, besides working towards a nice parsimonious class hierarchy, is that we can then modify Device to run its update() calls as tasks/threads/whatever, so we can separate the sensor reading (and even servo writing!) from the main logic. And we would only need to change Device in order to have that available for all sensors and such. thoughts for the future i guess.